Add KeenableWebSearch: low-latency web search for voice agents (#4923)
pipecat-ai/pipecat: changelog/4923.added.md

> - Added `KeenableWebSearch` (`pipecat.services.keenable.search`), an optional service that gives voice agents live web search and page reading via a hosted MCP server powered by [Keenable AI](https://keenable.ai). It exposes `search_web` (with optional `site` and date-range filters) and `fetch_page` function tools — pass `search.tools()` to your `LLMContext` and the LLM auto-registers their handlers (no `register_function` call, no async setup). The connection opens lazily on first use. Install with the `keenable` extra. Works keyless (`pro` mode); pass `api_key=` for higher rate limits and the lower-latency `realtime` mode (requires an enabled account), selected with `mode="realtime"`.

Reviewer: ```suggestion
- Added `KeenableWebSearch` (`pipecat.services.keenable.search`), an optional service that gives voice agents live web search and page reading via a hosted MCP server powered by [Keenable AI](https://keenable.ai). It exposes `search_web` (with optional `site` and date-range filters) and `fetch_page` function tools — pass `search.tools()` to your `LLMContext`. The connection opens lazily on first use. Install with the `keenable` extra. Works keyless (`pro` mode); pass `api_key=` for higher rate limits and access to the lower-latency `realtime` mode (requires an enabled account), selected with `mode="realtime"`.
```

Removed mention of a historical (albeit recent) mechanism change unrelated to this work, as well as an implementation route not taken, neither of which are meaningful to the reader.

Also made a small clarity fix.

Reviewer: One outstanding legibility problem: very unclear what the "pro mode" parenthetical means...

Author: Fixed the changelog.

About pro mode—the docs don't say anything. I'm going off of the original PR and just trying to port the meaning. So, yes, it is confusing. I asked the Keenable team to take a look, hoping that they'd find issues. They just said that it looks good without feedback though.

Reviewer: I understand now after having read the rest of the PR. "pro" is just the default mode, the opposite of "realtime". The parenthetical can probably simply be omitted from the above, to avoid unnecessary confusion.
